### DISCUSSION TITLE
Creazione nuovo cliente con API v2

### DISCUSSION THREAD

**Discussion by cleoni** [fattureincloud/api#67] [Q&A] [answered]
Salve,
sto cercando di creare nel cloud un nuovo oggetto di tipo "clients", e sto usando PHP

la documentazione dice di usare:

Create Client
`POST /c/{company_id}/entities/clients`
Creates a new client.

dove trovo un esempio di codice PHP che faccia questa post? Ho cercato nella documentzione senza trovarlo

**Comment by valmoz** [comment, marked as answer]
Buongiorno,
nella documentazione non abbiamo esempi dettagliati per tutti i metodi che esponiamo, però i metodi condividono una struttura standard e quindi è possibile seguire l'esempio di altri metodi per comprendere come procedere.

Nel caso della libreria PHP, può vedere esempi di utilizzo dei metodi API all'interno dei [file di test](https://github.com/fattureincloud/fattureincloud-php-sdk/blob/master/test/Api/ClientsApiTest.php), anche se i test utilizzano dei valori casuali all'interno dei body.

Per vedere lo schema può utilizzare la [API Reference](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/entities/clients)

Grazie
Mauro

**Reply by erphp01** [reply]
Buongiorno,
anche io ho lo stesso problema...
Sto passando dalla versione V1 alla V2 per un singolo progetto quindi sto optando per la versione con collegamento manuale e token che non scade.

Seguendo le guide sono riuscito senza problemi a fare ogni tipo di richiesta per cercare e visualizzare dati già presenti su fattureincloud.

Non riesco però a trovare un codice che mi dica come inviare i dati tramite guzzlehttp in modalità manuale. Ho provato ad aggiungere nel request una voce "data" senza successo... ottengo sempre {"error":{"message":"Invalid request.","validation_result":{"data":["The data field is required."]}}}

$url = "https://api-v2.fattureincloud.it/c/{company_id}/entities/clients";
$request = array("data"=>array(
	"name"=>"Test Prova",
	"type"=>"person",
	"first_name"=>"Test",
	"last_name"=>"Prova",
	"tax_code"=>"BLTGNI5ABCDA794E",
	"address_street"=>"Via Roma, 1",
	"address_postal_code"=>"20900",
	"address_city"=>"Milano")	
);
try {
  $response = $guzzlehttp->request('POST', $url, [
    'headers' => [
      'Accept' => 'application/json',
      'Authorization' => 'Bearer ' . $token
    ],
	'data' => json_encode($request)
  ]);
    $risposta= json_decode($response->getBody(),true);
} catch (ClientException $e) {
    echo Psr7\Message::toString($e->getRequest());
    echo Psr7\Message::toString($e->getResponse());
}

**Reply by emajo** [reply]
Ciao @erphp01 `'data' => json_encode($request)` è errato, guzzle richiede il che il body venga settato in questo modo `"json" => $request,`, di seguito il codice corretto:
```
<?php
$url = "https://api-v2.fattureincloud.it/c/{company_id}/entities/clients";
$request = [
    "data" => [
        "name" => "Test Prova",
        "type" => "person",
        "first_name" => "Test",
        "last_name" => "Prova",
        "tax_code" => "BLTGNI5ABCDA794E",
        "address_street" => "Via Roma, 1",
        "address_postal_code" => "20900",
        "address_city" => "Milano",
    ],
];
try {
    $response = $guzzlehttp->request("POST", $url, [
        "headers" => [
            "Accept" => "application/json",
            "Authorization" => "Bearer " . $token,
        ],
        "json" => $request,
    ]);
    $risposta = json_decode($response->getBody(), true);
} catch (ClientException $e) {
    echo Psr7\Message::toString($e->getRequest());
    echo Psr7\Message::toString($e->getResponse());
}
```

**Reply by erphp01** [reply]
Grazie mille!
Funziona tutto perfettamente.
Se posso permettermi suggerisco di inserire qualche esempio di inserimento in modalità manuale nella guida.
